src/Sarif/ExtensionMethods.cs

```diff
@@ -585,5 +586,15 @@ internal static void ConsumeElementOfDepth(this XmlReader xmlReader, int endElem
                 xmlReader.Read();
             }
         }
+
+        public static bool Add<T>(this ConcurrentDictionary<string, T> concurrentDictionary, string key, T value)
```

Comment:
>Add [](http://example.com/codeflow?start=27&length=3)

Why did you add this? What does the CurrentDictionary IDictionary.Add implementation call into? A non-thread-safe base method?

Reply:
just checked the source code of CurrentDictionary,
the IDictionary<TKey, TValue>.Add implementation
did call the same method as the TryAdd
so we can use it thread safely.

Also agree that we should not make the Add has a return value people will get confused. 

updated the code.